pkg/operator/ceph/cluster/osd/topology/topology_test.go

```diff
@@ -48,16 +48,23 @@ func TestCleanTopologyLabels(t *testing.T) {
 		"topology.rook.io/rack":             "r.rack",
 		"topology.rook.io/row":              "r.row",
 		"topology.rook.io/datacenter":       "d.datacenter",
+		"topology.rook.io/room":             "test",
+		"topology.rook.io/chassis":          "test",
+		"topology.rook.io/pod":              "test",
 	}
 	topology, affinity := ExtractOSDTopologyFromLabels(nodeLabels)
-	assert.Equal(t, 6, len(topology))
+	assert.Equal(t, 7, len(topology))
 	assert.Equal(t, "r-region", topology["region"])
 	assert.Equal(t, "z-zone", topology["zone"])
 	assert.Equal(t, "host-name", topology["host"])
 	assert.Equal(t, "r-rack", topology["rack"])
 	assert.Equal(t, "r-row", topology["row"])
 	assert.Equal(t, "d-datacenter", topology["datacenter"])
-	assert.Equal(t, "topology.rook.io/rack=r.rack", affinity)
+	assert.Equal(t, "topology.rook.io/chassis=test", affinity)
+	assert.Equal(t, "test", topology["chassis"])
+	assert.Equal(t, "", topology["pod"])
+	assert.Equal(t, "", topology["room"])
+
```

Comment:
Instead of modifying the Test**Clean**TopologyLabels, let's create a new unit test to test multiple kinds of duplication. This code is pretty complex for something so simple, and we want to make sure we exercise all the cases to ensure there aren't errors. 

Consider that there are 5 tests for the code that currently exists.

Reply:
Actually a this test does test all the end cases
https://github.com/rook/rook/blob/cb4133e7a5d7ae75ca26ddc109a3678965f8391b/pkg/operator/ceph/cluster/osd/topology/topology_test.go#L63
But if you say I can refactor it a little to sub test tasks(t.Run(){ ).